### PR TITLE
Add Integration Tests for Get/Set Client ID

### PR DESCRIPTION
**Pull request by hazen**
- Also remove useless response to set client Id
- Add and new make targets: `integration-tests`, `valgrind-tests`, `gdb-tests` and `lldb-tests`

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -192,3 +190,17 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = riak-c-client-async.pc \
 			riak-c-client.pc
 
+integration-tests: $(check_PROGRAMS)
+	./riak_c_cunit --integration
+
+valgrind-tests: $(check_PROGRAMS)
+	$(LIBTOOL) --mode=execute valgrind --leak-check=full --show-leak-kinds=all ./riak_c_cunit
```
Comment (bookshelfdave) [inline Makefile.am:197]: do you want another target like `valgrind-integration-tests`?

**Comment by bookshelfdave** [comment]
```
passed
  Test: test_integration_delete ...passed
  Test: test_integration_async_delete ...passed
  Test: test_integration_clientid ...passed
  Test: test_integration_async_clientid ...passed

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites      7      7    n/a      0        0
               tests    113    113    113      0        0
             asserts   2829   2829   2829      0      n/a
```

+1

File: Makefile.am (hunk 1 above)
Comment (bookshelfdave) [inline Makefile.am:197]: I'm not in front of a Linux box right now, but `make valgrind-tests` fails with an invalid option:

```
$ make valgrind-tests
/bin/sh ./libtool --mode=execute valgrind --leak-check=full --show-leak-kinds=all ./riak_c_cunit
valgrind: Bad option: --show-leak-kinds=all
```

Does this work for you on Linux?

Comment (hazen) [inline Makefile.am:197]: Works on Ubuntu 12.04LTS.  If you do a `brew update` and `brew install valgrind` then you'll get valgrind 3.9.0 on OSX as well.  The option is valid then, but fails in a horrible way:

```
valgrind: m_libcfile.c:73 (Int vgPlain_safe_fd(Int)): Assertion 'newfd >= VG_(fd_hard_limit)' failed.
==79590==    at 0x23803B57B: ???
==79590==    by 0x23803B52F: ???
==79590==    by 0x23803B876: ???
==79590==    by 0x238041064: ???
==79590==    by 0x2380406CD: ???
==79590==    by 0x23804063B: ???

sched status:
  running_tid=0
```

Comment (hazen) [inline Makefile.am:197]: Added `valgrind-integration-tests`

**Pull request merged**